Purchases/Caching/RCDeviceCache.m

```diff
@@ -56,48 +63,73 @@ - (void)clearCachesForAppUserID:(NSString *)appUserID
     [self.userDefaults removeObjectForKey:RCLegacyGeneratedAppUserDefaultsKey];
     [self.userDefaults removeObjectForKey:RCAppUserDefaultsKey];
     [self.userDefaults removeObjectForKey:[self purchaserInfoUserDefaultCacheKeyForAppUserID:appUserID]];
-    [self clearCachesTimestamp];
-    [self clearOfferings];
+    [self clearPurchaserInfoCacheTimestamp];
+    [self clearOfferingsCache];
 }
 
-- (BOOL)isCacheStale {
-    NSTimeInterval timeSinceLastCheck = -[self.cachesLastUpdated timeIntervalSinceNow];
-    return !(self.cachesLastUpdated != nil && timeSinceLastCheck < 60. * 5);
+#pragma mark - purchaserInfo
+
+- (nullable NSData *)cachedPurchaserInfoDataForAppUserID:(NSString *)appUserID
+{
+    return [self.userDefaults dataForKey:[self purchaserInfoUserDefaultCacheKeyForAppUserID:appUserID]];
 }
 
-- (void)resetCachesTimestamp
+- (void)cachePurchaserInfo:(NSData *)data forAppUserID:(NSString *)appUserID
 {
-    self.cachesLastUpdated = [NSDate date];
+    [self.userDefaults setObject:data
```

Comment:
We should probably synchronize these function for thread safety

Reply:
good call. an alternative (it might be out of scope for this PR) would be change the purchaser info object to be a dictionary that contains the info + the timestamp. That way we leverage thread safety from NSUserDefaults, instead of having separate fields that can fall out of sync.